sycl/source/detail/jit_device_binaries.hpp

```diff
@@ -124,6 +127,7 @@ class DeviceBinaryContainer {
   std::vector<_sycl_offload_entry_struct> PIOffloadEntries;
   std::vector<PropertySetContainer> PropertySets;
   std::vector<_sycl_device_binary_property_set_struct> PIPropertySets;
+  std::unique_ptr<char[]> CompileOptions;
```

Comment:
Why do we need to use a `unique_ptr` here instead of a `string`? Is it because of short-string optimization?

Reply:
No, I hadn't considered SSO tbh, and I think a `std::string` or `sycl::detail::string` would work just was well. I was following what the other helper structs in the file do to store their string members, e.g.:
https://github.com/intel/llvm/blob/a73f776a52a029a58c315aead126cdd2b662b6c1/sycl/source/detail/jit_device_binaries.hpp#L40

Reply:
Update: Tried it out, you're right, we can't use `std::string` because we'd `std::move` the `DeviceBinaryContainer` after we'd get the pointer. `sycl::detail::string` doesn't do SSO, but for consistency I'd keep the `unique_ptr<char[]>` here, ok?